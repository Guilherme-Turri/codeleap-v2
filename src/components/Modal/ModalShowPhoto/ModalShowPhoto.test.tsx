import { ModalShowPhoto } from '.'
import { componentRender } from '../../../render/componentRender'
import { screen, fireEvent } from '@testing-library/react'

describe('<ModalShowPhoto/>', () => {
  const ModalShowPhotoProps = {
    handleCancel: jest.fn(),
    avatarPic: '123'
  }

  it('should render on screen properly on desk', () => {
    const { container } = componentRender(<ModalShowPhoto {...ModalShowPhotoProps} />)
    const btn = screen.getByTestId('cancelbtn')
    const img = screen.getByTestId('img')
    expect(btn).toHaveStyleRule('top', '-20px')
    expect(btn).toHaveStyleRule('right', '-20px')
    expect(img).toHaveStyleRule('width', '450px')
    expect(img).toHaveStyleRule('height', '450px')
    expect(container).toMatchSnapshot();
  })

  it('should call handleCancel when close button is clicked', () => {
    componentRender(<ModalShowPhoto {...ModalShowPhotoProps} />)
    const btn = screen.getByTestId('cancelbtn')
    fireEvent.click(btn);
    expect(ModalShowPhotoProps.handleCancel).toHaveBeenCalled();
  })
  it('should render on screen properly on Mobile', () => {
    window.innerWidth = 450
    componentRender(<ModalShowPhoto {...ModalShowPhotoProps} />)
    const btn = screen.getByTestId('cancelbtn')
    const img = screen.getByTestId('img')
    expect(btn).toHaveStyleRule('top', '65px')
    expect(btn).toHaveStyleRule('right', '15px')
    expect(img).toHaveStyleRule('width', '350px')
    expect(img).toHaveStyleRule('height', '350px')
  })
})