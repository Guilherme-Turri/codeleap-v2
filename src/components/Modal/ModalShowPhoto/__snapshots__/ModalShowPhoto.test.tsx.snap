// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalShowPhoto/> should render on screen properly on desk 1`] = `
.c3 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c2 {
  color: #00CBA9;
  width: 100%;
  height: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
}

.c1 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 60px;
  height: 60px;
  border-radius: 50%;
  top: -20px;
  right: -20px;
  cursor: pointer;
}

.c4 {
  width: 450px;
  height: 450px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 50%;
  object-fit: cover;
  margin-top: 0.938rem;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      data-testid="cancelbtn"
    >
      <div
        class="c2"
        data-testid="Close"
      >
        <svg
          aria-hidden="true"
          class="c3"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            data-name="Layer 2"
          >
            <path
              d="M12 2a10 10 0 1 0 10 10A10 10 0 0 0 12 2zm2.71 11.29a1 1 0 0 1 0 1.42 1 1 0 0 1-1.42 0L12 13.41l-1.29 1.3a1 1 0 0 1-1.42 0 1 1 0 0 1 0-1.42l1.3-1.29-1.3-1.29a1 1 0 0 1 1.42-1.42l1.29 1.3 1.29-1.3a1 1 0 0 1 1.42 1.42L13.41 12z"
              data-name="close-circle"
            />
          </g>
        </svg>
      </div>
    </div>
    <img
      alt="Profile"
      class="c4"
      data-testid="img"
      src="123"
    />
  </div>
</div>
`;
